src/Services/SubmitService/SubmitService.cs

```diff
@@ -166,5 +171,57 @@ public async Task<string> PaymentSubmission(MappingEntity mappingEntity, string
 
             throw new ApplicationException($"SubmitService::PaymentSubmission, An exception has occured when response content from {postUrl} is null, Gateway responded with {response.StatusCode} status code, Message: {JsonConvert.SerializeObject(response)}");
         }
+
+        public async Task<string> RedirectSubmission(MappingEntity mappingEntity, string form, string sessionGuid)
+        {
+            if (_submissionServiceConfiguration.FakeSubmission)
+                return ProcessFakeSubmission(mappingEntity, form, sessionGuid, string.Empty);
+
+            var currentPage = mappingEntity.BaseForm.GetPage(_pageHelper, mappingEntity.FormAnswers.Path);
+
+            var postUrl = currentPage.GetSubmitFormEndpoint(mappingEntity.FormAnswers, _environment.EnvironmentName.ToS3EnvPrefix());
+
+            var redirectEntity = _actionHelper.GenerateUrl(postUrl.RedirectUrl, mappingEntity.FormAnswers);
+
+            if (string.IsNullOrEmpty(postUrl.URL))
+            {
+                throw new ApplicationException($"SubmitService::RedirectSubmission, No submission URL has been provided for FORM: {form}, ENVIRONMENT: {_environment.EnvironmentName}");
+            }
+
+            if (string.IsNullOrEmpty(redirectEntity.Url))
+            {
+                throw new ApplicationException($"SubmitService::RedirectSubmission, No redirect URL has been provided for FORM: {form}, ENVIRONMENT: {_environment.EnvironmentName}");
+            }
+
+            if (string.IsNullOrWhiteSpace(postUrl.AuthToken))
```

Comment:
My Resharper suggested this as replacement code for this if/else, I think it looks good myself but see what you think. If you do change it, could you also change the same bit of code in the other methods to match?

_gateway.ChangeAuthenticationHeader(string.IsNullOrWhiteSpace(postUrl.AuthToken)
                ? string.Empty
                : postUrl.AuthToken);